frontend/src/stores/knowledges/getters.ts

```diff
@@ -1,8 +1,23 @@
 import { computed } from 'vue'
 import { knowledges } from '@/stores/knowledges/state'
+import type { Role } from '@/domain/Role'
+import type { Knowledge } from '@/domain/Knowledge'
+import { get } from '@/services/http'
 
 const getKnowledge = computed(() => {
   return knowledges?.value
 })
 
-export { getKnowledge }
+async function getKnowledgesByRoleId(roleId: number): Promise<Knowledge[]> {
+  if (knowledges?.value?.length > 0) {
+    return knowledges.value
```

Comment:
Here you are returning all knowledges, with no criteria. But we just want the knowledges of a given roleId. You should filter your available knowledges by roleId here